sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/trace/XraySamplerClient.java

```diff
@@ -3,37 +3,82 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+// Includes work from:
+
+/*
+ * Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Portions copyright 2006-2009 James Murty. Please see LICENSE.txt
+ * for applicable license terms and NOTICE.txt for applicable notices.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
 package io.opentelemetry.sdk.extension.aws.trace;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonParser;
 import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
 import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.module.SimpleModule;
 import io.opentelemetry.sdk.extension.aws.internal.JdkHttpClient;
+import java.io.IOException;
 import java.io.UncheckedIOException;
+import java.math.BigDecimal;
 import java.util.Collections;
+import java.util.Date;
 import java.util.Map;
 
 final class XraySamplerClient {
 
-  private static final ObjectMapper OBJECT_MAPPER =
-      new ObjectMapper()
-          .setSerializationInclusion(JsonInclude.Include.NON_EMPTY)
-          // In case API is extended with new fields.
-          .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, /* state= */ false);
+  private static final ObjectMapper OBJECT_MAPPER;
+
+  static {
```

Comment:
Out of curiosity - why separate static init block instead of in-place, as it as was previously?